lib/Fetcher/FeedFetcher.php

```diff
@@ -321,10 +332,28 @@ protected function buildFeed(FeedInterface $feed, string $url, bool $getFavicon,
         }
         $newFeed->setAdded($this->time->getTime());
 
-        if (!$getFavicon) {
-            return $newFeed;
+        $favicon = $feed->getLogo();
+
+        // check if feed has a logo
+        if (!is_null($favicon)) {
+            $favicon_path = join("/", [$this->ITempManager->getTempBaseDir(), basename($favicon)]);
+            file_put_contents(
+                $favicon_path,
+                file_get_contents($url)
+            );
+            $is_image = substr(mime_content_type($favicon_path), 0, 5) === "image";
+            // check if file is actually an image
+            if ($is_image) {
+                list($width, $height, $type, $attr) = getimagesize($favicon_path);
+                // check if image is square else fall back to favicon
+                if ($width !== $height) {
+                    $favicon = $this->faviconFactory->get($url);
+                }
+            }
+        } else {
+            $favicon = $this->faviconFactory->get($url);
         }
```

Comment:
Can you split this into a function? It has very little to do with the feed fetching and I think it's less readable.

Reply:
True makes sense, will do

Reply:
created a function to replace that big block